src/clients/HubPoolClient.ts

```diff
@@ -768,11 +768,21 @@ export class HubPoolClient extends BaseAbstractClient {
       searchConfig: eventSearchConfigs.map(({ eventName, searchConfig }) => ({ eventName, searchConfig })),
     });
     const timerStart = Date.now();
-    const [currentTime, pendingRootBundleProposal, ...events] = await Promise.all([
-      this.hubPool.getCurrentTime({ blockTag: searchConfig.toBlock }),
-      this.hubPool.rootBundleProposal({ blockTag: searchConfig.toBlock }),
-      ...eventSearchConfigs.map((config) => paginatedEventQuery(this.hubPool, config.filter, config.searchConfig)),
+
+    const { hubPool } = this;
+    const multicallFunctions = ["getCurrentTime", "rootBundleProposal"];
+    const [multicallOutput, ...events] = await Promise.all([
+      hubPool.callStatic.multicall(
+        multicallFunctions.map((f) => hubPool.interface.encodeFunctionData(f)),
+        { blockTag: searchConfig.toBlock }
+      ),
+      ...eventSearchConfigs.map((config) => paginatedEventQuery(hubPool, config.filter, config.searchConfig)),
     ]);
+    const [currentTime, pendingRootBundleProposal] = multicallFunctions.map((fn, idx) => {
+      const output = hubPool.interface.decodeFunctionResult(fn, multicallOutput[idx]);
+      return BigNumber.isBigNumber(output) ? output.toNumber() : spreadEvent(output);
```

Comment:
Why is this conversion needed? Doesn't seem like that was happening in the previous version.

Reply:
Isn't this just because `multicall` returns `bytes[]` and not the deserialized types?

Reply:
Sorry, I'm talking about the conversion of line 783. The one on 782 makes sense (decoding the multicall).